test/mocha/unit/traffic-switch/traffic-switcher-i-token-fee-fetcher.test.ts

```diff
@@ -0,0 +1,292 @@
+import sinon, { SinonSpy } from 'sinon'
+import { metric } from '@uniswap/smart-order-router/build/main/util/metric'
+import { ProviderConfig } from '@uniswap/smart-order-router/build/main/providers/provider'
+import { TrafficSwitcherITokenFeeFetcher } from '../../../../lib/util/traffic-switch/traffic-switcher-i-token-fee-fetcher'
+import { GraphQLTokenFeeFetcher } from '../../../../lib/graphql/graphql-token-fee-fetcher'
+import { OnChainTokenFeeFetcher, TokenFeeMap } from '@uniswap/smart-order-router/build/main/providers/token-fee-fetcher'
+import { BigNumber } from 'ethers'
+import { expect } from 'chai'
+import { MetricLoggerUnit } from '@uniswap/smart-order-router'
+import { TrafficSwitcher } from '../../../../lib/util/traffic-switch/traffic-switcher'
+
+describe('TrafficSwitcherITokenFeeFetcher', () => {
+  let spy: SinonSpy
+
+  const methodFetchFeesReturnFee100 = async (addresses: string[], _?: ProviderConfig): Promise<TokenFeeMap> => {
```

Comment:
could we add some tests to the base class as well? :p

Reply:
base class is abstract, so this is the actual testing it mocking the TrafficSwitcherITokenFeeFetcher implementation.
Happy to add more tests here though

Reply:
yeah, I was thinking of just creating an implementation TestTrafficSwitcher to test the methods of the base class in isolation, under the assumption that the `TokenFeeFetcher` traffic switch will be removed at the end of the experiment, when that happens then the base class will not have any tests.